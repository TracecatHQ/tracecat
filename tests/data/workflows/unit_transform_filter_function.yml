title: Reshape data
description: Test that we can use a forwarder to reshape data
entrypoint:
  ref: filter_empty
inputs: #2D list
  matrix:
    - [1, 2, 3]
    - [4, 5, 6]
    - []
    - [7, 8, 9]
    - [10, 11, 12]
    - []

actions:
  - ref: filter_empty
    action: core.transform.forward
    args:
      value: ${{ FN.filter(INPUTS.matrix, 'len(x) > 0') }}
