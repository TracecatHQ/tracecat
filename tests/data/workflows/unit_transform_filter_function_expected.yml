ACTIONS:
  filter_empty:
    result: [[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 11, 12]]
    result_typename: list

INPUTS:
  matrix:
    - [1, 2, 3]
    - [4, 5, 6]
    - []
    - [7, 8, 9]
    - [10, 11, 12]
    - []
TRIGGER:
